ydb/core/blobstorage/dsproxy/dsproxy_get_impl.cpp

```diff
@@ -35,6 +36,47 @@ void TGetImpl::PrepareReply(NKikimrProto::EReplyStatus status, TString errorReas
             outResponse.LooksLikePhantom = PhantomCheck
                 ? std::make_optional(false)
                 : std::nullopt;
+
+            if (IntegrityCheck) {
+                const TBlobState &blobState = Blackboard.GetState(query.Id);
+                outResponse.CompletenessFailed = !blobState.HasWrittenQuorum(*Info, nullptr);
+
+                std::unordered_set<ui32> partIndexes;
+                ui32 partCount = 0;
+                for (const auto &item : blobState.PartMap) {
+                    if (!item.Data.IsEmpty()) {
+                        partIndexes.insert(item.PartIdRequested);
+                        ++partCount;
+                    }
+                }
+                TRope data;
+                if (Info->Type.GetErasure() == TErasureType::Erasure4Plus2Block && partIndexes.size() > 3 || 
+                    Info->Type.GetErasure() != TErasureType::Erasure4Plus2Block && partCount > 1) {
```

Comment:
There are also exist another erasure types, which in current implementation will be treated as block4+2. It can lead to errors. I would recommend to explicitly branch such cases and do nothing